### ISSUE TITLE
Roompool does not provide rooms.

### ISSUE THREAD

**Issue by yncyrydybyl** [polychatproject/polychat-appservice#10]
It looks like the rooms are created. But they are not provides when the user wants to join.

**Comment by yncyrydybyl** [comment]
```
Feb 07 21:31:05 alpha matrix-appservice-polychat[2287089]: room.event {"age":90,"content":{"displayname":"testuser","membership":"invite"},"event_id":"$b8vWaTRm-clBaAwzQFwzuThhdegiIqJJMFG_o8Cd5Xc","origin_server_ts":1707337865123,"room_id":"!wJjFuPFQIZuJSOavzp:polychat.de","sender":"@polychat:polychat.de","state_key":"@testuser:polychat.de","type":"m.room.member","unsigned":{"age":90},"user_id":"@polychat:polychat.de"}
Feb 07 21:31:05 alpha matrix-appservice-polychat[2287089]: Appservice - - - [07/Feb/2024:20:31:05 +0000] "PUT /_matrix/app/v1/transactions/20356? HTTP/1.1" - - "-" "Synapse/1.99.0"
Feb 07 21:31:05 alpha matrix-appservice-polychat[2287089]: API: Created Polychat !wJjFuPFQIZuJSOavzp:polychat.de
Feb 07 21:31:05 alpha matrix-appservice-polychat[2287089]: Appservice Processing transaction 20357
Feb 07 21:31:05 alpha matrix-appservice-polychat[2287089]: Appservice Processing event of type m.room.power_levels
Feb 07 21:31:05 alpha matrix-appservice-polychat[2287089]: room.event {"age":75,"content":{"ban":50,"events":{"m.room.avatar":50,"m.room.canonical_alias":50,"m.room.encryption":100,"m.room.history_visibility":100,"m.room.name":50,"m.room.power_levels":100,"m.room.server_acl":100,"m.room.tombstone":100},"events_default":0,"historical":100,"invite":0,"kick":50,"redact":50,"state_default":50,"users":{"@polychat:polychat.de":100,"@testuser:polychat.de":50},"users_default":0},"event_id":"$J--Yn7szCzm0jfVOkneYP7dCjU9rgaHWpJfbFZZHBfM","origin_server_ts":1707337865230,"prev_content":{"ban":50,"events":{"m.room.avatar":50,"m.room.canonical_alias":50,"m.room.encryption":100,"m.room.history_visibility":100,"m.room.name":50,"m.room.power_levels":100,"m.room.server_acl":100,"m.room.tombstone":100},"events_default":0,"historical":100,"invite":0,"kick":50,"redact":50,"state_default":50,"users":{"@polychat:polychat.de":100},"users_default":0},"replaces_state":"$wLi13IzWRX19lR1Mb0UmUdPZHil4v-fVgjxAnxKV6rg","room_id":"!wJjFuPFQIZuJSOavzp:polychat.de","sender":"@polychat:polychat.de","state_key":"","type":"m.room.power_levels","unsigned":{"age":75,"prev_content":{"ban":50,"events":{"m.room.avatar":50,"m.room.canonical_alias":50,"m.room.encryption":100,"m.room.history_visibility":100,"m.room.name":50,"m.room.power_levels":100,"m.room.server_acl":100,"m.room.tombstone":100},"events_default":0,"historical":100,"invite":0,"kick":50,"redact":50,"state_default":50,"users":{"@polychat:polychat.de":100},"users_default":0},"prev_sender":"@polychat:polychat.de","replaces_state":"$wLi13IzWRX19lR1Mb0UmUdPZHil4v-fVgjxAnxKV6rg"},"user_id":"@polychat:polychat.de"}
Feb 07 21:31:05 alpha matrix-appservice-polychat[2287089]: Appservice - - - [07/Feb/2024:20:31:05 +0000] "PUT /_matrix/app/v1/transactions/20357? HTTP/1.1" - - "-" "Synapse/1.99.0"
Feb 07 21:31:05 alpha matrix-appservice-polychat[2287089]: API: Requested Polychat !wJjFuPFQIZuJSOavzp:polychat.de
Feb 07 21:31:05 alpha matrix-appservice-polychat[2287089]: API: Requested Polychat !wJjFuPFQIZuJSOavzp:polychat.de
Feb 07 21:31:37 alpha matrix-appservice-polychat[2287089]: API: Requested Polychat !wJjFuPFQIZuJSOavzp:polychat.de
Feb 07 21:31:37 alpha matrix-appservice-polychat[2287089]: API: Requested Polychat !wJjFuPFQIZuJSOavzp:polychat.de
Feb 07 21:31:54 alpha matrix-appservice-polychat[2287089]: API: Requested Polychat !wJjFuPFQIZuJSOavzp:polychat.de
Feb 07 21:31:54 alpha matrix-appservice-polychat[2287089]: API: Requested Polychat !wJjFuPFQIZuJSOavzp:polychat.de
Feb 07 21:31:57 alpha matrix-appservice-polychat[2287089]: API: Requested Polychat !wJjFuPFQIZuJSOavzp:polychat.de
Feb 07 21:31:57 alpha matrix-appservice-polychat[2287089]: API: Requested Polychat !wJjFuPFQIZuJSOavzp:polychat.de
Feb 07 21:32:09 alpha matrix-appservice-polychat[2287089]: API: Requested Polychat !wJjFuPFQIZuJSOavzp:polychat.de
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]: API: Error claiming a sub room for !wJjFuPFQIZuJSOavzp:polychat.de for telegram
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]: 156 |     if (!Array.isArray(unclaimedSubRoomsForThisNetwork)) {
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]: 157 |         throw Error('E_NO_SUB_ROOM_FOR_THIS_NETWORK');
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]: 158 |     }
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]: 159 |     const subRoomIndex = unclaimedSubRoomsForThisNetwork.findIndex(subRoom => subRoom.ready && subRoom.inviteUrl);
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]: 160 |     if (subRoomIndex === -1) {
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]: 161 |         throw Error('E_OUT_OF_SUB_ROOMS');
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]:                     ^
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]: error: E_OUT_OF_SUB_ROOMS
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]:       at /app/src/index.ts:161:15
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]:       at claimSubRoom (/app/src/index.ts:154:36)
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]:       at /app/src/api.ts:136:33
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]:       at /app/src/api.ts:100:90
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]:       at handle (/app/node_modules/express/lib/router/layer.js:95:5)
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]:       at next (/app/node_modules/express/lib/router/route.js:144:7)
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]:       at done (/app/node_modules/multer/lib/make-middleware.js:45:7)
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]:       at indicateDone (/app/node_modules/multer/lib/make-middleware.js:49:68)
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]:       at /app/node_modules/multer/lib/make-middleware.js:166:7
Feb 07 21:32:18 alpha matrix-appservice-polychat[2287089]:       at emitCloseNT (node:stream:1405:49)

```

**Comment by jaller94** [comment]
It does create the sub rooms but never marks them as ready because it does not see the inite link messages.

**Comment by jaller94** [comment]
Maybe fixed in f3d84e94925753b4960a2705ffad681617814b07.

I incorrectly tested for `msgtype === 'notice'` when it should have been `msgtype === 'm.notice'`. This is now slightly better covered by a unit test.

**Comment by jaller94** [comment]
Tested and fixed
